installers/rpm-code/codes.sh

```diff
@@ -208,44 +209,54 @@ codes_err() {
 }
 
 codes_install() {
-    local module=$1
+    declare module=$1
     shift
-    local args=( "$@" )
-    local prev=$(pwd)
-    local d=$HOME/src/radiasoft/codes/$module-$(date -u +%Y%m%d.%H%M%S)
+    declare args=( "$@" )
+    declare prev=$(pwd)
+    declare d=$HOME/src/radiasoft/codes/$module-$(date -u +%Y%m%d.%H%M%S)
     rm -rf "$d"
     mkdir -p "$d"
     codes_msg "Build: $module"
     codes_msg "Directory: $d"
     cd "$d"
-    local codes_module=$module
-    local -A codes_dir=()
+    declare codes_module=$module
+    declare -A codes_dir=()
     codes_dir_setup
     # Needed for pyenv
     install_source_bashrc
     install_script_eval "codes/$module.sh"
-    local f=${module}_main
+    declare f=${module}_main
     if ! codes_is_function "$f"; then
         install_error "function=$f not defined for code=$module"
     fi
     $f ${args[@]+"${args[@]}"}
     cd "$prev"
-    local p=${module}_python_install
+    declare p=${module}_python_install
     if codes_is_function "$p"; then
         codes_msg "Running: $p"
         cd "$d"
         "$p"
         codes_install_python_done
     fi
-    local d=${codes_dir[prefix]}/lib64
+    declare t=${module}_test
+    if codes_is_function "$t"; then
+        codes_msg "Running: $t"
+        # python adds cwd to PYTHONPATH so go to a path where
+        # cwd won't contain any python that would be added
+        mkdir -p /tmp/$(date -u +%Y%m%d.%H%M%S)
```

Comment:
@robnagler I want to call this out. Wasn't sure about a good location.

Reply:
Good idea. You can use install_tmp_dir which will clean it up better. Also it labels the directory a bit better so you know you can delete them manually. Finally, it uses /var/tmp which usually has more room than /tmp.

Reply:
7c8960c